### PR TITLE
Add the ability to use ReferenceArrayInput without child

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] Default `ReferenceArrayInput` child to `AutocompleteArrayInput`
- [x] Add `options` prop support in `SelectArrayInput`(as documented)
- [x] Rewrite the doc of all Array inputs 😓
- [x] Order the Field and Input navigation alphabetically

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/SelectArrayInput.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                    choices={choices}
+                    sx={{ width: 300 }}
+                    create={<CreateRole />}
+                />
+            </SimpleForm>
+        </Create>
+    </AdminContext>
+);
+
+export const EmptyText = () => (
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/SelectArrayInput.stories.tsx:103]: The name of this story is not related to what is showing, plus I don't get what is trying to show, it looks the same as the Basic story

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/DualListInput.md [hunk 2]
```diff
@@ -5,19 +5,298 @@ title: "The DualListInput Component"
...
+const possibleValues = ['programming', 'lifestyle', 'photography'];
+const ucfirst = name => name.charAt(0).toUpperCase() + name.slice(1);
+const choices = possibleValues.map(value => ({ id: value, name: ucfirst(value) }));
+
+<DualListInput source="roles" choices={choices} />
+```
+
+## `disableValue`
+
+By default, `<DualListInput>` renders the choices with the field `disabled: true` as disabled.
```
Comment (WiXSL) [inline docs/DualListInput.md:145]:
```suggestion
By default, `<DualListInput>` renders the choices with the field `disabled` as disabled.
```

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/ReferenceArrayInput.md [hunk 3]
```diff
@@ -7,103 +7,332 @@ title: "The ReferenceArrayInput Component"
...
+| `label`            | Optional | `string`                                    | -                                  | Useful only when `ReferenceArrayInput` is in a Filter array, the label is used as the Filter label.                 |
 | `page`             | Optional | `number`                                    | 1                                  | The current page number                                                                                             |
 | `perPage`          | Optional | `number`                                    | 25                                 | Number of suggestions to show                                                                                       |
-| `reference`        | Required | `string`                                    | ''                                 | Name of the reference resource, e.g. 'posts'.                                                                       |
+| `queryOptions`     | Optional | [`UseQuery Options`](https://tanstack.com/query/v4/docs/reference/useQuery?from=reactQueryV3&original=https://react-query-v3.tanstack.com/reference/useQuery)| `{}`  | `react-query` client options     |
```
Comment (WiXSL) [inline docs/ReferenceArrayInput.md:104]: Actually, this is not true, `queryOptions` prop is part of the `ReferenceInput` but not yet of the `ReferenceArrayInput`

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: docs/ReferenceArrayInput.md [hunk 4]
```diff
@@ -7,103 +7,332 @@ title: "The ReferenceArrayInput Component"
...
+
+You can override this default by specifying the `optionText` prop in the child component. For instance, for an `<AutocompleteArrayInput>`:
+
+```jsx
+<ReferenceArrayInput source="tags_ids" reference="tags">
+    <AutocompleteArrayInput optionText="reference" />
+</ReferenceArrayInput>
+```
+
+## `queryOptions`
```
Comment (WiXSL) [inline docs/ReferenceArrayInput.md:274]: Actually, this is not true, queryOptions prop is part of the ReferenceInput (https://github.com/marmelab/react-admin/pull/8192) but not yet of the ReferenceArrayInput

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx [hunk 5]
```diff
@@ -130,7 +130,7 @@ export const CheckboxGroupInput: FunctionComponent<CheckboxGroupInputProps> = pr
         );
     }
 
-    if (!isLoading && allChoices === undefined) {
+    if (!isLoading && !fetchError && allChoices === undefined) {
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx:133]: Maybe worth doing this in the `RadioButtonGroupInput` and `SelectInput` components as well

File: packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx [hunk 6]
```diff
@@ -183,7 +183,7 @@ export const CheckboxGroupInput: FunctionComponent<CheckboxGroupInputProps> = pr
         [allChoices, formOnChange, formOnBlur, optionValue, value]
     );
 
-    if (isLoading && allChoices.length === 0) {
+    if (isLoading && (!allChoices || allChoices.length === 0)) {
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx:186]: Same

File: packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx [hunk 7]
```diff
@@ -219,7 +219,7 @@ export const CheckboxGroupInput: FunctionComponent<CheckboxGroupInputProps> = pr
                 />
             </FormLabel>
             <FormGroup row={row}>
-                {allChoices.map(choice => (
+                {allChoices?.map(choice => (
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx:222]: Same

**Review by WiXSL — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx (hunk 5 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/CheckboxGroupInput.tsx:133]: I've tested that it's not necessary for these
